File: Sliding-Window/2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        cur = 0

        # Count initial 'W's in the first window of size k
        for i in range(k):
            if blocks[i] == 'W':
                cur += 1

        minCount = cur
        start = 0

        # Sliding window technique
        for i in range(k, len(blocks)):
            if blocks[i] == 'W':
                cur += 1
            if blocks[start] == 'W':
                cur -= 1
            start += 1

            minCount = min(minCount, cur)

        return minCount

File: Sliding-Window/2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int cur = 0;

        for(int i = 0; i < k; i++)
            if(blocks[i] == 'W') cur++;
        cout << cur;

        int minCount = cur, start = 0;
        for(int i = k; i < blocks.size(); i++){
            if(blocks[i] == 'W') cur++;
            if(blocks[start] == 'W') cur--;
            start++;

            minCount = min(minCount, cur);
        }

        return minCount;
    }
};